client/v2/tx/config.go

```diff
@@ -0,0 +1,68 @@
+package tx
+
+import (
+	txsigning "cosmossdk.io/x/tx/signing"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	signingtypes "github.com/cosmos/cosmos-sdk/types/tx/signing"
+)
+
+type TxConfig interface {
+	GetTxParams() TxParameters
+	GetSignConfig() TxSigningConfig
+}
+
+type TxSigningConfig interface {
+	SignModeHandler() *txsigning.HandlerMap
+	SigningContext() *txsigning.Context
+	MarshalSignatureJSON([]signingtypes.SignatureV2) ([]byte, error)
+	UnmarshalSignatureJSON([]byte) ([]signingtypes.SignatureV2, error)
+}
+
+type TxParameters struct {
+	timeoutHeight uint64
+	chainID       string
+	memo          string
+	signMode      signingtypes.SignMode
+
+	AccountConfig
+	GasConfig
+	FeeConfig
+	ExecutionOptions
+	ExtensionOptions
+}
```

Comment:
'big' change here. I've splitted all the fields in v1 into specific structs for reutilization and easier reading.

Reply:
@julienrbrt what do you think of this particular change?